src/program.cpp

```diff
@@ -352,7 +355,10 @@ std::vector<argument> generic_eval(const module* mod,
                             }));
         }
         assert(results.find(ins) != results.end());
-        assert(results.at(ins).get_shape() == ins->get_shape());
+        if(not ins->get_shape().dynamic())
```

Comment:
They will not be the same since the instruction shape will be dynamic but the result shape will be fixed. With the current way the code is structured, any operator's `compute()` function will have a fixed output shape.